spec/routemaster/integration/cache_spec.rb

```diff
@@ -50,6 +53,14 @@
           .to({ field: 'test'}.to_json)
       end
 
+      it "sets the event_index onto the cache" do
```

Comment:
I appreciate that the specs here aren't fabulous — but this, I think, is a smell. Exactly how we deal with concurrency and version our cache should not leak, which means `EventIndex` should most definitely _not_ appear in integration tests.

If anything, as long as it does it's impossible to confidently refactor those internals.

Ideally, we'd do something like simulate concurrent requests that return different responses, and detect whether we've actually cached the latest data or not?